src/uiScreens.c

```diff
@@ -191,6 +219,48 @@ void ui_displayAmountScreen(
 	);
 }
 
+void ui_displayUint64Screen(
+        const char* screenHeader,
+        uint64_t value,
+        ui_callback_fn_t callback
+)
+{
+	char valueStr[30];
+	str_formatUint64(value, valueStr, SIZEOF(valueStr));
+
+	ui_displayPaginatedText(
+	        screenHeader,
+	        valueStr,
+	        callback
+	);
+}
+
+void ui_displayValidityBoundaryScreen(
+        const char* screenHeader,
+        uint64_t boundary,
+        uint8_t networkId, uint32_t protocolMagic,
+        ui_callback_fn_t callback
+)
+{
+	char boundaryStr[30];
+	explicit_bzero(boundaryStr, SIZEOF(boundaryStr));
+
+	if ((networkId == MAINNET_NETWORK_ID) && (protocolMagic == MAINNET_PROTOCOL_MAGIC)) {
```

Comment:
this probably deserves an explanatory comment that we know the epoch/slot configuration reliably only for mainnet